Python/pylifecycle.c

```diff
@@ -1665,6 +1671,7 @@ flush_std_files(void)
 static void
 finalize_interp_types(PyInterpreterState *interp)
 {
+    _PyGenericAlias_Fini(interp);
```

Comment:
```suggestion
```

Then add it to 

```c

static void
finalize_interp_clear(PyThreadState *tstate)
{
    int is_main_interp = _Py_IsMainInterpreter(tstate->interp);

    _PyExc_ClearExceptionGroupType(tstate->interp);```

    _PyGenericAlias_Fini(tstate->interp);
```

Reply:
@sobolevn Looks like this has not to been applied yet?

Reply:
Second thoughts, I don't really know where is the correct place to call `_PyGenericAlias_Fini`.

I have the instinct that the destructor of this cache will execute any code. We should clear this cache as early as possible.